lib/class-wp-theme-json-gutenberg.php

```diff
@@ -956,6 +956,27 @@ public function get_settings() {
 		}
 	}
 
+	/**
+	 * Processes the CSS, to apply nesting.
+	 *
+	 * @param string $css      The CSS to process.
+	 * @param string $selector The selector to nest.
+	 *
+	 * @return string The processed CSS.
+	 */
+	private function process_nested_css( $css, $selector ) {
+		$processed_css = '';
+
+		// Split CSS nested rules.
+		$parts = explode( '&', $css );
+		foreach ( $parts as $part ) {
+			$processed_css .= ( ! str_contains( $part, '{' ) )
+				? $selector . '{' . $part . '}' // If the part doesn't contain braces, it applies to the root level.
+				: $selector . $part; // Prepend the selector, which effectively replaces the "&" character.
+		}
+		return $processed_css;
+	}
```

Comment:
It would be good to have unit tests for this new method. We can't test private methods, but we can check the `$theme_json->get_stylesheet()` output. See `test_get_stylesheet_handles_custom_css` for examples.

Would this method be applicable outside of block custom CSS? Maybe we should use a more explicit name for it. What do you think?

Reply:
> Would this method be applicable outside of block custom CSS? Maybe we should use a more explicit name for it. What do you think?

I can't imagine where this would be used outside the block custom-css... I'll rename it to something more specific and write some unit tests 👍

Reply:
method renamed to `process_blocks_custom_css_nested_styles`, and PHPUnit tests added 👍